### ISSUE TITLE
button / checkbox wigdets cannot be mapped to Bela's digital IO

### ISSUE THREAD

**Issue by giuliomoro** [grame-cncm/faust#234]
architecture/bela.cpp:
```
     virtual void addButton(const char* label, FAUSTFLOAT* zone) { skip(); }
     virtual void addCheckButton(const char* label, FAUSTFLOAT* zone) { skip(); }
```
Is there any  good reason for this?

It's a shame because they would be great for the digitals. Current workaround is to use 

```
gate = hslider("h:Basic_Parameters/gate [1][tooltip:noteOn = 1, noteOff = 0][BELA:DIGITAL_0]",1,0,1,1);
```
instead of 
```
gate = button("h:Basic_Parameters/gate [1][tooltip:noteOn = 1, noteOff = 0][BELA:DIGITAL_0]") : int;
```

**Comment by sletz** [comment]
I think it was just left out. Can you possibly implement them and prepare a PR for that ?

**Comment by giuliomoro** [comment]
if you have time, that'd be brilliant: I am not really familiar with the FAUST API. Then I can verify the result on the board.

**Comment by sletz** [comment]
Ok. Is hust dis that in https://github.com/grame-cncm/faust/commit/8a5275b1940311fd9ac2b060300948721556037a

basically branching the button/checkbutton with low = 0 and hi = 1 ([low hi] is the range of the controller). How do you plan to connect button/checkbutton from external controllers ?

**Comment by giuliomoro** [comment]
great, that works. Closed 8a5275b1940311fd9ac2b060300948721556037a

**Comment by giuliomoro** [comment]
> How do you plan to connect button/checkbutton from external controllers ?

What do you mean? Right now this could be anything that you plug into one of the Bela GPIOs, e.g.: a tactile switch, a SPST switch, a slide selector ...
